frontend/src/components/Month/Dates.tsx

```diff
@@ -0,0 +1,38 @@
+import moment, { now } from "moment";
+
+type Dtype = {
+  month: number;
+  date: number;
+  day: number;
+};
+
+export const getDates = (criteria: moment.Moment) => {
+  const startDate = criteria.clone();
+  const dates: Dtype[][] = new Array(6);
+
+  for (let i = 0; i < dates.length; i++) dates[i] = new Array();
+
+  for (let dayCnt = 1; dayCnt <= startDate.day(); dayCnt++) {
+    const now = startDate.clone().subtract(dayCnt, "d");
+
+    dates[0].unshift({
+      month: now.month() + 1,
+      date: now.date(),
+      day: now.day()
+    });
+  }
```

Comment:
여기서, `startDate`를 매번 clone하지 않고 아래와 같이,
```suggestion
  const preDate = startDate.clone();
  for (let dayCnt = 1; dayCnt <= startDate.day(); dayCnt++) {
    preDate.subtract(1, "d");

    dates[0].unshift({
      month: preDate.month() + 1,
      date: preDate.date(),
      day: preDate.day()
    });
  }
```
하나의 변수(`preDate`)에 복사본을 저장해서 사용할 수 있습니다!